CIScripts/Test/Tests/vRouterAgentService.Tests.ps1

```diff
@@ -0,0 +1,119 @@
+Param (
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+)
+
+. $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+. $PSScriptRoot\..\..\Common\VMUtils.ps1
+. $PSScriptRoot\..\PesterHelpers\PesterHelpers.ps1
+
+. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+$TestConf = Get-TestConfiguration
+$Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
+
+Describe "vRouter Agent service" {
+    Context "enabling" {
+        It "is enabled" {
+            Get-AgentServiceStatus -Session $Session `
+                | Should Be "Running"
+        }
+
+        BeforeEach {
+            Enable-AgentService -Session $Session
+        }
+    }
+    
+    Context "disabling" {
+        It "is disabled" {
+            Get-AgentServiceStatus -Session $Session `
+                | Should Be "Stopped"
+        }
+
+        It "does not restart" {
+            Consistently {
+                Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
+            }
+        }
+
+        BeforeEach {
+            Enable-AgentService -Session $Session
+            Disable-AgentService -Session $Session
+        }
+    }
+
+    Context "given vRouter Forwarding Extension is NOT running" {
+        It "crashes" {
+            Eventually {
+                Read-SyslogForAgentCrash -Session $Session -After $BeforeCrash `
+                    | Should Not BeNullOrEmpty
+            } -Duration 60
+        }
+
+        BeforeEach {
+            Disable-VRouterExtension -Session $Session `
+                -AdapterName $TestConf.AdapterName `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName
+            Test-IsVRouterExtensionEnabled -Session $Session `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName `
+                | Should Be $false
+            $BeforeCrash = Invoke-Command -Session $Session -ScriptBlock { Get-Date }
+            Enable-AgentService -Session $Session
+        }
+    }
+
+    Context "given vRouter Forwarding Extension is running" {
+        It "runs correctly" {
+            Get-AgentServiceStatus -Session $Session `
+                | Should Be "Running"
+        }
+
+        BeforeEach {
+            Test-IsVRouterExtensionEnabled -Session $Session `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName `
+                | Should Be $true
+            Enable-AgentService -Session $Session
+        }
+    }
+    
+    Context "vRouter Forwarding Extension was disabled while Agent was running" {
+        It "crashes" {
+            Eventually {
+                Read-SyslogForAgentCrash -Session $Session -After $BeforeCrash `
+                    | Should Not BeNullOrEmpty
+            }
```

Comment:
It's using timeout = 3 here, which may be not enough.

Reply:
Every time in `vRouter Forwarding Extension was disabled while Agent was running, it crashes` this timeout was enough. `given vRouter Forwarding Extension is NOT running, it crashes` however needed a longer timeout. I would leave it short for now, increase it only if it's needed.